### ISSUE TITLE
Add Album art to Audio files

### ISSUE THREAD

**Issue by MonsterSe7en** [jeanslack/Videomass#352]
## Thanks for taking the time to submit a feature request!   

**Make sure you are using the most recent version of Videomass**   
Perhaps your feature request has already been implemented. The most recent version can be downloaded from Github   

# Thank you
And sir. I want to thank you for taking the time to develop this program. It's super useful

Would be useful if you could add an option to add an album art (An image) to audio files. I had to make a batch script to do it. 

> [!note]
> @echo off
>for %%i in (*.m4a) do 
>(
>    ffmpeg -i "%%i" -i "album_art.jpg" -c copy -map 0 -map 1 -disposition:v:0 attached_pic "%%~ni_temp.m4a"
>   move /y "%%~ni_temp.m4a" "%%i"
>)

I tried making various edits to the Profile file but I couldn't get it to add the album art

![image](https://github.com/user-attachments/assets/437baee8-acd1-495c-b08f-461b2a70e517)

**Comment by jeanslack** [comment]
HI @MonsterSe7en 

You're on the right track: your command works, but it seems that the full path to the `"album_art.jpg"` file is missing.

Then you should complete the dialog forms (name, description, and output format (output audio format in this case).

**Comment by MonsterSe7en** [comment]
Yes, the name and dialog forms were completed. Didn't work, Videomass executes the commands but I don't get the intended result (The m4a files are generated but they don't have the album art)

And with regards to the output audio format, it generally populates with the container name I choose, I prefer removing that as the message says leave blank if you want a copy of source (I've tried with the field left populated and blank, no difference).

I'm assuming this is user error from my end or has something to do with the filepath not working well with videomass. 

Sharing a screenshot with the complete information

![image](https://github.com/user-attachments/assets/97a8ddf4-2245-4057-95cf-e3275573c74b)


But please check if you can add another field to choose an album art

**Comment by jeanslack** [comment]
> And with regards to the output audio format, it generally populates with the container name I choose, I prefer removing that as the message says leave blank if you want a copy of source (I've tried with the field left populated and blank, no difference).

Sure!

Also remove the `-vn` flag otherwise it will not include the image on the audio file. FFmpeg treats image files as video frames.

For completeness:
`-i "F:\Videos\Folder Name\Album Art.jpg" -map 0 -map 1 -c copy -disposition:v:0 attached_pic -sn -map_metadata 0`

**Comment by MonsterSe7en** [comment]
oh!! Thank you.

 -i "F:\Temp\IMG_20210301_232028_100.jpg" -map 0 -map 1 -c copy -disposition:v:0 attached_pic -sn -map_metadata 0
 
I tried, It renders the audio file (Which works fine) but no album art... 😢😢 

(But it works fine and gives me the expected result if I send the argument via Command line directly to ffmpeg with no issues)... 🧐

 ![image](https://github.com/user-attachments/assets/ccd38ecf-38e5-453d-ad2a-822f2f068b45)

Oh. Well... I appreciate your time and patience. Thank you

And I'll leave this classic meme that's plagued us coders for decades!!

![raf,750x1000,075,t,322e3f_696a94a5d4 u2](https://github.com/user-attachments/assets/d1a75637-6a35-4dbf-9f00-e42d071a98aa)

Cheers 
MonsterSe7en

**Comment by jeanslack** [comment]
I updated the presets by including some profiles to be able to add album art to audio files in mp3, m4a, mp4, flac, ogg formats. These profiles can be found in the `Utilities` preset from Videomass version 5.0.18, or you can download them individually [here](https://github.com/jeanslack/Videomass-presets/releases/tag/v0.24).

I hope they can be useful to you.
Cheers

**Comment by Happyclem** [comment]
Is there any way in Videomass to keep existing art-cover ? Maybe have a toggle in the "various" tab ? 
Even with "copy metadata", my file comes out with the cover.

EDIT ANSWER : In a profil, write
1st pass : 
`-map 0:a:? -map 0:v? -filter:a: loudnorm=I=-16.0:TP=-1.5:LRA=11.0:print_format=summary -sn -dn -f null
`
2nd pass : 
`-map 0:a? -map 0:v? -c:a copy -c:v copy -sn -dn -map_metadata 0`

**Comment by jeanslack** [comment]
@Happyclem 
Sorry, but I completely missed this post because it had already been closed.
Reopening.

**Comment by jeanslack** [comment]
> Is there any way in Videomass to keep existing art-cover ? Maybe have a toggle in the "various" tab ? Even with "copy metadata", my file comes out with the cover.

Yes, adding the `-c:v` flag should copy the art-cover as the image is recognized as a video stream by FFmpeg.
For this operation, you must use the **Preset Manager** because **Audio/Video Conversions** doesn't include this feature yet.

> EDIT ANSWER : In a profil, write 1st pass : `-map 0:a:? -map 0:v? -filter:a: loudnorm=I=-16.0:TP=-1.5:LRA=11.0:print_format=summary -sn -dn -f null ` 2nd pass : `-map 0:a? -map 0:v? -c:a copy -c:v copy -sn -dn -map_metadata 0`

Exatly.
